### PR TITLE
Add logging of applied adjust rules

### PR DESCRIPTION
**Pull request by happz**

### CODE REVIEW

**Comment by happz** [comment]
Blocked by https://github.com/teemtee/fmf/pull/197

**Review by psss — COMMENTED** [review]
Looks nice and very useful! We need to wait though until the `fmf` change is merged and released.

**Comment by thrix** [comment]
Moved to 1.28, as we need a fmf update, and it will take > a wekk

File: tmt/base.py
Comment (thrix) [inline tmt/base.py:350]: the UX looks weird to me, could we just color the outcome of the adjust rule apply maybe? And ideally without **, it looks odd, I believe it would be enought to mark the word:

I would leave the whole text in `cyan`, while:

* green - applied
* red - not applied
* blue (or white) - skipped

**Review by thrix — CHANGES_REQUESTED** [review]

**Review by janhavlin — APPROVED** [review]

**Comment by psss** [comment]
> > Looks nice and very useful! We need to wait though until the `fmf` change is merged and released.
> 
> @psss seems it is waiting now on you :)

The plan is to release the fresh `fmf` this week so that this one is unblocked.

File: tmt/base.py
Comment (thrix) [reply tmt/base.py:350]: @happz kind ping on this comment

Comment (happz) [reply tmt/base.py:350]: Colors are there to help on the console, they should not be the primary information carrier. I don't care about that enough to not change them.

I can also drop `**`, not crucial, but helpful in highlighting the decision when there are no colors, e.g. when the output is piped into a file. This, I believe, will be most of the time, as there's plenty of `adjust` around & searching a file is much easier than following the output on the terminal.

I don't understand whether changing colors and dropping `**` would make the UX less weird in your eyes, or whether is there anything else left to tweak.

**Review by therazix — APPROVED** [review]

File: tmt/base.py
Comment (psss) [reply tmt/base.py]: Agree with @thrix that `**` looks sort of disturbing and I'd say in non-color view search highlight helps much more. I propose to use `cyan` for keys instead of `green` as the green color almost everywhere causes it hard to actually see the green applied rules. @happz, are you ok with 5917b59b?

**Comment by psss** [comment]
Hm, seems that `fedora-39` jobs will be red until the distro is released (planned for Oct 17). What to do? Enable the `fmf` copr repo for `f39` jobs?

**Review by psss — APPROVED** [review]

**Pull request merged**
